src/RustyObjectStore.jl

```diff
@@ -81,52 +98,203 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
     return nothing
 end
 
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `request_timeout_secs::Option{Int}`: (Optional) Client request timeout in seconds.
+- `connect_timeout_secs::Option{Int}`: (Optional) Client connection timeout in seconds.
+- `max_retries::Option{Int}`: (Optional) Maximum number of retry attempts.
+- `retry_timeout_secs::Option{Int}`: (Optional) Maximum amount of time from the initial request after which no further retries will be attempted (in seconds).
+"""
+struct ClientOptions
+    params::Dict{String, String}
+
+    function ClientOptions(;
+        request_timeout_secs::Option{Int} = nothing,
+        connect_timeout_secs::Option{Int} = nothing,
+        max_retries::Option{Int} = nothing,
+        retry_timeout_secs::Option{Int} = nothing
+    )
+        params = Dict()
+        if !isnothing(request_timeout_secs)
+            params["timeout"] = "$(string(request_timeout_secs))s"
+        end
+
+        if !isnothing(connect_timeout_secs)
+            params["connect_timeout"] = "$(string(connect_timeout_secs))s"
+        end
+
+        if !isnothing(max_retries)
+            params["max_retries"] = string(max_retries)
+        end
+
+        if !isnothing(retry_timeout_secs)
+            params["retry_timeout_secs"] = string(retry_timeout_secs)
+        end
+
+        return new(params)
+    end
+end
+
+abstract type AsConfig end
+
+function as_config(wrapper::AsConfig) end
+
 """
     $TYPEDEF
 
 # Arguments
-$TYPEDFIELDS
+- `url::String`: Url of the object store container root path.
+  It must include the cloud specific url scheme (s3://, azure:// ...).
```

Comment:
is there a known fixed list of url schemes we support? (is it just `s3://`, `az://`, `azure://`?) 

if so, can we just list them all here?

Reply:
Sure!